worker/jobTypes/S3Publish.js

```diff
@@ -80,17 +83,20 @@ class S3PublishClass {
       `make stage`
     ];
 
-    // the way we now build is to search for a specific function string in worker.sh
-    // which then maps to a specific target that we run
-    const workerContents = fs.readFileSync(`repos/${this.GitHubJob.getRepoDirName()}/worker.sh`, { encoding: 'utf8' });
-    const workerLines = workerContents.split(/\r?\n/);
+    // check if worker.sh exists
+    if (fs.existsSync(`repos/${this.GitHubJob.getRepoDirName()}/worker.sh`)) {
+      // the way we now build is to search for a specific function string in worker.sh
+      // which then maps to a specific target that we run
+      const workerContents = fs.readFileSync(`repos/${this.GitHubJob.getRepoDirName()}/worker.sh`, { encoding: 'utf8' });
```

Comment:
I would refactor this -- too much code duplication. Can we put the common code in a function and call the function/check for null on response and if not null run the loop.